src/gluonts/transform.py

```diff
@@ -345,14 +347,25 @@ class FlatMapTransformation(Transformation):
     def __call__(
         self, data_it: Iterator[DataEntry], is_train: bool
     ) -> Iterator:
+        num_idle_transforms = 0
         for data_entry in data_it:
+            num_idle_transforms += 1
             try:
                 for result in self.flatmap_transform(
```

Comment:
So, `num_idle_transforms` is increased, iff `flat_map_transform` returns an empty result?

Do you have an example when that is the case?

Reply:
Yes, I ran into this. 

If you have a dataset with time series that all have length 100 and you set prediction length to something larger than 100. The transformations loop forever. The `InstanceSplitter` skips over time series that are too short. This is the right thing when there are time series with varying length.

Reply:
I think I understand. Since we loop over the dataset to sample enough data, this leads to problems if there are no applicable time-series.

I wonder if we can filter first and then check if any data is availabe.

Reply:
Mmh, maybe. One problem is that the filtering is stochastic during training. Every time you loop over the dataset you may get a different number of results. 

We could perhaps do one loop over the dataset (without repetition) and make sure we get at least one output or something like that. But there are also edge cases her. You could have a dataset with just one time series. Because of the stochastic filtering, you may not yield anything in this test execution and so we would need more complicated logic.